src/dmd/typesem.d

```diff
@@ -363,6 +363,9 @@ private void resolveHelper(TypeQualified mt, const ref Loc loc, Scope* sc, Dsymb
         s = sm.toAlias();
     }
 
+    if (auto ed = s.isEnumDeclaration())
+        s.dsymbolSemantic(sc);
```

Comment:
This adds an odd special case, #12438 seems to be the more general approach.

Reply:
I admit that doing semantic here isn't ideal, however, I think it's better to analyze the enum in its entirety. In contrast #12438 does semantic on the first member only and defers the rest for later.

That being said, I have no problem conceding as long as the issue is fixed.

Reply:
> I admit that doing semantic here isn't ideal, however, I think it's better to analyze the enum in its entirety. In contrast #12438 does semantic on the first member only and defers the rest for later.

Why? the specs says clearly
* If the EnumBaseType is not explicitly set, and the first EnumMember has an AssignExpression, it is set to the type of that AssignExpression. Otherwise, it defaults to type int.

Reply:
A self-contained solution as a compromise of both PR's:

```patch
diff --git a/src/dmd/denum.d b/src/dmd/denum.d
index 267fb4f01..9886d85dc 100644
--- a/src/dmd/denum.d
+++ b/src/dmd/denum.d
@@ -342,6 +342,8 @@ extern (C++) final class EnumDeclaration : ScopeDsymbol
                 Loc locx = loc.isValid() ? loc : this.loc;
                 memtype = memtype.typeSemantic(locx, _scope);
             }
+            else
+                dsymbolSemantic(this, _scope);
         }
         if (!memtype)
         {
```

Allthough this is slightly less capable w.r.t. to deep forward references.